### PR TITLE
Use structopt

### LINKED ISSUE #94: Improve argument parsing
Currently the argument parsing is handled by `Clap` and is far from optimal.

Consider removing to something that reduces the clutter would be preferable, i.e. `structopt`.

### PR DESCRIPTION
**Pull request by montekki**
Fixes #94 

Move to `structopt` but keep the existing cmdline interface for compatibility. So this gets a bit tricky since old `enum` type cannot be turned into flags by `structopt`, only into subcommands. Needs a bit more testing for now though.

### CODE REVIEW

**Comment by montekki** [comment]
Interestingly the LOC is not reduced.

**Comment by drahnr** [comment]
Readability improved significantly though

File: src/cmdline.rs [hunk 1]
```diff
@@ -101,35 +101,33 @@ impl TryFrom<Command2> for Command {
...
-            ));
+            ))
         } else {
             let Command2 { cmd, .. } = cmd;
 
             let mut cmd = cmd.into_iter();
 
             if let Some(exe) = cmd.next() {
-                let cmdline = cmd.map(|s| s.to_owned()).collect::<Vec<_>>();
+                let cmdline = cmd.map(|s| s).collect::<Vec<_>>();
```
Comment (drahnr) [inline src/cmdline.rs:130]: `map(|s| s)` seems a bit pointless?

Comment (montekki) [reply src/cmdline.rs]: yeah, just dumbly fixed the clippy warn

**Comment by drahnr** [comment]
Could we rebase this and get this into master? I'd rather not see the work go stale!

**Review by drahnr — APPROVED** [review]

**Pull request merged**